src/vm/exceptionhandling.cpp

```diff
@@ -4663,6 +4664,11 @@ VOID DECLSPEC_NORETURN DispatchManagedException(PAL_SEHException& ex)
                     // Unwind to the managed caller of the helper
                     PAL_VirtualUnwind(&frameContext, NULL);
                 }
+#ifdef _TARGET_ARM_
```

Comment:
I think that we can get rid of this change and also of the adjustment for marked JIT helpers above if we do the adjustment in the HandleHardwareException directly to the context in the exception.
Let me make and test such change for the marked jit helpers and if that works as expected, you can update your PR. I'll try to do the change right away and let you know about the result.